extension/test/e2e/credit-card-3ds-native.spec.js

```diff
@@ -0,0 +1,178 @@
+const nodeStatic = require('node-static')
+const iTSetUp = require('../integration/integration-test-set-up')
+const ctpClientBuilder = require('../../src/ctp/ctp-client')
+const { routes } = require('../../src/routes')
+const httpUtils = require('../../src/utils')
+const { assertPayment, createPaymentWithOriginKeyResponse, initPuppeteerBrowser } = require('./e2e-test-utils')
+const MakePaymentFormPage = require('./pageObjects/CreditCardMakePaymentFormPage')
+const RedirectPaymentFormPage = require('./pageObjects/RedirectPaymentFormPage')
+const CreditCardNativePage = require('./pageObjects/CreditCard3dsNativePage')
+
+// Flow description: https://docs.adyen.com/checkout/3d-secure/native-3ds2/web-component
+describe('::creditCardPayment3dsNative::', () => {
+  let browser
+  let ctpClient
+
+  // See more: https://docs.adyen.com/development-resources/test-cards/test-card-numbers
+  const creditCards = [
+    {
+      name: 'Mastercard',
+      creditCardNumber: '5454 5454 5454 5454'
+    },
+    {
+      name: 'Visa',
+      creditCardNumber: '4917 6100 0000 0000'
+    }
+  ]
+
+  beforeEach(async () => {
+    const fileServer = new nodeStatic.Server()
+    routes['/make-payment-form'] = async (request, response) => {
+      fileServer.serveFile('./test/e2e/fixtures/3ds-v2-make-payment-form.html', 200, {}, request, response)
+    }
+    routes['/redirect-payment-form'] = async (request, response) => {
+      fileServer.serveFile('./test/e2e/fixtures/redirect-payment-form.html', 200, {}, request, response)
+    }
+    routes['/return-url'] = async (request, response) => {
+      const body = await httpUtils.collectRequestData(request)
+      return httpUtils.sendResponse({
+        response,
+        headers: {
+          'Content-Type': 'text/html'
+        },
+        data: '<!DOCTYPE html><html><head></head>'
+          + `<body><div id=redirect-response>${body[0].toString()}</div></body></html>`
+      })
+    }
+
+    ctpClient = ctpClientBuilder.get()
+    await iTSetUp.initServerAndExtension({ ctpClient, routes, testServerPort: 8080 })
+    browser = await initPuppeteerBrowser()
+  })
+
+  afterEach(async () => {
+    await iTSetUp.stopRunningServers()
+    await browser.close()
+  })
+
+  creditCards.forEach(({
+                         name, creditCardNumber, creditCardDate = '03/30', creditCardCvc = '737'
+                       }) => {
+    // eslint-disable-next-line no-template-curly-in-string
+    it(`when credit card issuer is ${name} and credit card number is ${creditCardNumber}, ` +
+      'it should successfully finish the payment with 3DS native authentication flow',
+      async () => {
+        const baseUrl = process.env.API_EXTENSION_BASE_URL
```

Comment:
Refactoring point: All those baseUrls should be managed by config.load.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/321/commits/3357ae9869afff108125e98633761970346470d9